File: TwoPointer/javascript/11659.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const [n, m] = input[0].split(" ").map(Number);
const arr = input[1]
  .split(" ")
  .map(Number)
  .reduce(
    (acc, cur, idx) => {
      acc.push(acc[idx] + cur);
      return acc;
    },
    [0]
  );
let result = "";
for (let i = 2; i < input.length - 1; i++) {
  let [a, b] = input[i].split(" ").map(Number);

  result += arr[b] - arr[--a];
  result += "\n";
}

console.log(result);

File: TwoPointer/python/11659.py
import sys
input = sys.stdin.readline
n,m = map(int,input().split())
arr = list(map(int,input().split()))
sum_arr = [0]
for i in range(len(arr)):
  sum_arr.append(arr[i] + sum_arr[i])
for _ in range(m):
  a,b = map(int,input().split())
  a -= 1
  print(sum_arr[b] - sum_arr[a])
